distro/rpm/openvpn.init.d.rhel

```diff
@@ -224,17 +224,24 @@ case "$1" in
 	fi
 	;;
   status)
-	if [ -f $lock ]; then
-	    for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
-		if [ -s $pidf ]; then
-		    kill -USR2 `cat $pidf` >/dev/null 2>&1
-		fi
-	    done
-	    echo "Status written to /var/log/messages"
-	else
-	    echo "openvpn: service not started"
-	    exit 1
-	fi
+        if [ -f $lock ]; then
+            for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
+                PID=$(cat $pidf)
+                if [ -e /proc/$PID ]; then
+                    echo "openvpn is running. pid: $PID"
+                    exit 0
+                else
+                    echo "openvpn is dead but lock file exists: $lock and pid file exists: $pidf"
+                    exit 1
+                fi
+            done 
```

Comment:
Having struggled with a few ScientificLinux 6  boxes over the years with several configs, I can understand this pain.  Generally speaking I have not used `service openvpn status` on my SL6/CentOS5 boxes precisely due to this issue. 

However, I would recommend not to do the exit immediately.  Loop through all configs, and report the status for each configuration file.  If one of the configs failed, then do `exit 1`.

By the way, I see this patch set uses exit 1, 2 and 3.  Are those standard exit values for upstart and/or traditional SysV init scripts?

And a final note.  Who will ensure these fixes gets included into the distro packages?  It seems every package manager likes to re-invent wheels (as we've seen with systemd too).  I'd like to change that though, but things needs to be followed up.  Otherwise we have nice init scripts in our source tree which nobody uses.